File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SumOfSquares(n int) int {
	result := 0
	for i := 1; i <= n; i++ {
		result += i * i
	}
	return result
}

func SquareOfSums(n int) int {
	result := 0
	for i := 1; i <= n; i++ {
		result += i
	}
	return result * result
}

func Difference(n int) int {
	return SquareOfSums(n) - SumOfSquares(n)
}

File: typescript/difference-of-squares/difference-of-squares.ts
export default class Squares {
  sumOfSquares = 0
  squareOfSums = 0
  difference = 0

  constructor(n: number) {
    let sums = 0
    for (let i = 1; i <= n; i++) {
      sums += i
      this.sumOfSquares += Math.pow(i, 2)
    }
    this.squareOfSums = Math.pow(sums, 2)
    this.difference = this.squareOfSums - this.sumOfSquares
  }
}